### DISCUSSION TITLE
Any better ways to display @RequestMapping endpoints?

### DISCUSSION THREAD

**Discussion by ksg97031** [owasp-noir/noir#334] [General]
Hi everyone,

When it comes to Java Spring's `@RequestMapping`, it supports multiple HTTP methods by default. 
As a result, when I define an endpoint like `@RequestMapping("hello")`, it appears as five different endpoints:

- GET /hello
- POST /hello
- DELETE /hello
- PUT /hello
- PATCH /hello

This raises a concern about the increased amount of duplicated output. 
Are there any better ways to display these `@RequestMapping` endpoints to avoid redundancy? For example, displaying them in a single line ("GET", "POST", "DELETE", "PUT", "PATCH") or using a different notation?

Thanks for your insights!

**Comment by hahwul** [comment]
@ksg97031 
I personally prefer to adhere to standard practice, which involves representing different methods for the same path as separate endpoints. However, if I need to use a more concise expression, I would opt for this approach.

```
ANY /hello
```

Since the `ANY` method is used in some proxies (just like [AWS API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-method-settings-method-request.html)), I thought a concise approach might also be recognizable to users. However, I believe that it is better to express each method separately, even if they may yield similar outcomes frequently 😄

**Reply by ksg97031** [reply]
Thanks a lot for your feedback!

**Reply by hahwul** [reply]
Hiya @ksg97031 👋🏼
I've had some time to think it over. 
If we add this feature, it would be a good idea to include an OUTPUT or CONFIG flag similar to...

* `--use-unify-method`
* `--use-any-method`

What do you think? I think it would be great if we could give users the choice!

**Reply by ksg97031** [reply]
> Hiya @ksg97031 👋🏼 I've had some time to think it over. If we add this feature, it would be a good idea to include an OUTPUT or CONFIG flag similar to...
> 
> * `--use-unify-method`
> * `--use-any-method`
> 
> What do you think? I think it would be great if we could give users the choice!

Oh! I also think that providing options would be a good choice for users. 👍

Additionally, we need to determine the scope for unifying methods. 
If the same code and line have multiple methods, it seems better to unify them.

For RequestMapping, we can use the `ANY` keyword as you suggested. However, `ANY` might be too broad. 
If there are two or more methods like `GET` and `POST` on the same endpoint, should we handle them differently? Or should we ignore such cases?"

**Comment by hahwul** [comment]
We recently talked about this topic face-to-face.
The design is more complex than anticipated, but if possible, it will be added as a feature soon :)
